### PR TITLE
[W4.3d][W11-A2]Zhan Yu

### PR DESCRIPTION
**Pull request by ZY-Ang**
Rebased branch for LO-ImplementClasses. Sorry for previous crap PRs

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Format: `help`
 ### Adding a person: `add`
 Adds a person to the address book<br>
 Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Address Format: `BLOCK, STREET, UNIT, POSTAL CODE`
```
Comment (okkhoy) [inline doc/UserGuide.md]: :+1: for updating docs!

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,11 +8,15 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Sesame Street, #03-21A, 690123";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in 'BLOCK, STREET, UNIT, POSTAL CODE' format";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: this line is quite long. try to stick to 120 characters

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -21,36 +25,50 @@
...
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public boolean isValidAddress(String addressString) throws IllegalValueException {
+        String[] addressData = getAddressData(addressString);
+        if (addressData.length != 4) {
+            return false;
+        }
+
+        block = new Block(addressData[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: What does our coding standard say about using numbers in array indices? [Hint! read magic numbers]

File: src/seedu/addressbook/data/person/Block.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Represents the block in a person's address.
+ */
+public class Block {
+    private final String value;
+    private static final String BLOCK_VALIDATION_REGEX = "^[a-zA-Z0-9]*$";
+    private static final String MESSAGE_BLOCK_CONSTRAINTS = "Person address block must be in an alphanumeric format without special characters";
+
+    /**
+     * Block Constructor
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: you may skip header comments for trivial methods like this one

File: test/input.txt [hunk 5]
```diff
@@ -40,19 +40,43 @@
...
+  add invalid address man p/9584731 e/add@add.add a/4, 32, invalid address lane, #05-34, 542345
+  add invalid address man p/9584731 e/add@add.add a/432, invalid address lane, #05-34, Concordia, 542345
+  
+  # should display invalid block format
+  add invalid block man p/9584731 e/block@block.block a/9$9, invalid block street, #05-34, 542345
+  add invalid block man p/9584731 e/block@block.block a/9'9, invalid block street, #05-34, 542345
+  add invalid block man p/9584731 e/block@block.block a/9 9, invalid block street, #05-34, 542345
+  
+  # should display invalid street format
+  add invalid street man p/9584731 e/street@street.street a/99, ha-ha-ha, #05-34, 542345
```
Comment (okkhoy) [inline test/input.txt]: good to see you have updated the tests as well!

**Comment by okkhoy** [comment]
@ZY-Ang 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Address.java (hunk 3 above)
Comment (ZY-Ang) [reply src/seedu/addressbook/data/person/Address.java]: Yeah, crap, need to abstract it (maybe into a constant in Address) and make it readable.

Comment (ZY-Ang) [reply src/seedu/addressbook/data/person/Address.java]: 7767ef7ba490767ed8cca8511d16ac933035376e

File: src/seedu/addressbook/data/person/Address.java (hunk 2 above)
Comment (ZY-Ang) [reply src/seedu/addressbook/data/person/Address.java]: 👍 52f50a29499c5694c350fc0a2ab264da415cd469

**Pull request closed without merging**